docs/IR.md

```diff
@@ -157,6 +157,8 @@ Graphs have the following properties:
 name|string|The name of the model graph.
 node|Node[]|A list of nodes, forming a partially ordered computation graph based on input/output data dependencies.
 initializer|Tensor[]|A list of named tensor values. When an initializer has the same name as a graph input, it specifies a default value for that input. When an initializer has a name different from all graph inputs, it specifies a constant value.
+sparse_initializer|SparseTensor[]|A list of initializer tensor values that are represented by sparse tensors.
+graph_initializer|Graph[]|A list of initializers that are represented by graphs, which describe how to compute the initializer tensor. Each initializer graph has zero inputs and one output, which acts as the initializer tensor. The initializer graph name acts as the initializer name.
```

Comment:
I don't quite sure the reasons behind graph_initializers. If an initializer is input-dependent, it should be computed from the input in the main computation graph. If an initializer is input-independent, it sounds a constant and the current initializer support is sufficient. Are there cases we must compute initializers from separated graphs? Or this new field is mainly for convenience?

ONNX training spec will support some ways to **compute** the initializers and possibly have a similar semantic here. It's also strange if we have a main graph, training graph, and initialization graph --- should we execute initialization graph each time we execute the training graph?

Reply:
I see. I created the PR after we saw that the training spec does not include an initialization graph. Training is the main purpose for this change.

IMO there may still be other reasons to keep the initialization separate from training. For instance, if for inference some weights are represented by large tensors that can be simply computed, it would be better to initialize tensors once. Currently, if they are added as constant initializers, the file size would be large. If they are added as nodes to the graph, they will be recomputed every time, which might not be efficient. 

It would probably be fine if this change will only be applied in `TrainingInfoProto`. What do you think?

Reply:
If we want to use graph to compute large constant tensor, we can specify that logic in the main inference graph (that is, we insert **nodes** into inference graph). Is it necessary to define another field for that logic?

For combining this proposal and training, we need to first list out the scenarios we want to support. For example, if we only want to express `initialization + training` in the training graph, we can merge `initialization` and `training` into one single graph and extra fields are not necessary.

Reply:
@wschin I am understanding this PR is trying to add a field to show the initialization logic of weights in training scenario. @tbennum correct me if I'm wrong. This is why I think this should align with training proposal.

Reply:
@linkerzhang yes, that is the goal.

Reply:
Training's initialization is a very specific process. It should be executed for training from scratch but it can't be executed for transferring learning. If we put some a layer's weight into `graph_initializer`, it means we don't necessarily have a placeholder for the tensor, which can cause some problems when training algorithm updates its value. For example, if I define `W` as a graph initializer, where should I save its value produced by the training algorithm?

The signature of `graph_initializer` looks fine. We just need to refine its semantic. It should be an optional step to update `existing` initializers rather than defining new initializers.